src/duckling/game/collision/collision-types.service.ts

```diff
@@ -64,7 +64,9 @@ export class CollisionTypesService {
     }
 
     addCollisionType(collisionType : string) {
-        this._store.dispatch(_collisionTypesAction(Array.from(this.collisionTypes.getValue().concat([collisionType]))));
+        let newCollisionTypes = new Set<string>(this.collisionTypes.getValue());
```

Comment:
I would change _registerMultipleCollisionTypes so that it takes an array. You could replace this implementation with 
```
addCollisionTypes(collisionType : string) {
   this._registerMultipleCollisionTypes([collisionType]);
}
```